roles/kubernetes-apps/ansible/templates/coredns-deployment.yml.j2

```
@@ -38,21 +38,9 @@ spec:
         {{ dns_extra_tolerations | list | to_nice_yaml(indent=2) | indent(8) }}
 {% endif %}
       affinity:
-        podAntiAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-          - topologyKey: "kubernetes.io/hostname"
-            labelSelector:
-              matchLabels:
-                k8s-app: kube-dns{{ coredns_ordinal_suffix }}
-        nodeAffinity:
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 100
-            preference:
-              matchExpressions:
-              - key: node-role.kubernetes.io/control-plane
-                operator: In
-                values:
-                - ""
+{% if coredns_affinity is defined %}
+        {{ coredns_affinity | to_nice_yaml(indent=2) | indent(8) }}
+{% endif %}
```

You don't need the condition if you have defined `coredns_affinity` in `roles`.

```suggestion
        {{ coredns_affinity | to_nice_yaml(indent=2) | indent(8) }}
```

---

I agree